addons/block_code/inspector_plugin/block_script_inspector.gd

```diff
@@ -11,7 +16,7 @@ func _parse_begin(object):
 	var block_code := object as BlockCode
 
 	var button := Button.new()
-	button.text = "Open Block Script"
+	button.text = tr("Open Block Script")
```

Comment:
Should this be `TxUtils.translate("Open Block Script")`? If not, why not?

Reply:
This is a great question. The documentation for [Object.tr](https://docs.godotengine.org/en/latest/classes/class_object.html#class-object-method-tr) hasn't been updated since `TranslationDomain` was added, but essentially it's a wrapper around `translate`. Prior to 4.4, that would be [TranslationServer.translate](https://docs.godotengine.org/en/latest/classes/class_translationserver.html#class-translationserver-method-translate). In 4.4, that would be [TranslationDomain.translate](https://docs.godotengine.org/en/latest/classes/class_translationdomain.html#class-translationdomain-method-translate) after looking up the [Object's translation domain](https://docs.godotengine.org/en/latest/classes/class_object.html#class-object-method-get-translation-domain). There are a couple reasons not to call `translate` directly:

1. `tr` and `tr_n` respect [Object.can_translate_messages](https://docs.godotengine.org/en/latest/classes/class_object.html#class-object-method-can-translate-messages).
2. The GDScript POT extractor will extract constant strings passed to `tr` and `tr_n` as translatable strings. Now I'm noticing that `TxUtils.translate("constant string")` won't do that and I'll need to add a hack for that.

The only reason you'd want to call `translate` directly is if you're in a static context since you can't use `Object.tr` in that case.

Reply:
OK, thanks for the clear explanation!

Reply:
I think I should probably document this. Often times I forget that after I've gone deep on something and it becomes clear to _me_, it's still likely unclear for someone else.

Reply:
I added some info in the README about this.